### ISSUE TITLE
Telegraf/Prometheus - inputs.mongodb not reporting replica set metrics

### ISSUE THREAD

**Issue by chezneUE** [influxdata/telegraf#4680]
### Relevant telegraf.conf:

https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mongodb

### System info:

[Include Telegraf version, operating system name, and other relevant details]

Telegraf 1.7 latest
CentOS7

user created for mondoDB: 
db.createUser({user: "telegraf", pwd: "telegraf", roles: [{role:"userAdminAnyDatabase",db:"admin"}]})

### Steps to reproduce:

1. Follow documentation setup for output to prometheus and include inputs.mongodb setup following documentation

2. per database stats works, user can access tables etc, MongoDB and Telegraf latest stable version 

### Expected behavior:
reporting the following at least display 0

repl_deletes_per_sec (integer)
repl_getmores_per_sec (integer)
repl_inserts_per_sec (integer)
repl_lag (integer)
repl_queries_per_sec (integer)
repl_updates_per_sec (integer)
repl_oplog_window_sec (integer)

### Actual behavior:

Missing no information

### Additional info:

log doesn't return anything unusual, all is operational

2018-09-11T20:31:10Z D! Attempting connection to output: prometheus_client
2018-09-11T20:31:10Z D! Successfully connected to output: prometheus_client
2018-09-11T20:31:10Z I! Starting Telegraf v1.7.4
2018-09-11T20:31:10Z I! Loaded inputs: inputs.cpu inputs.kernel inputs.swap inputs.mongodb inputs.disk inputs.diskio inputs.mem inputs.processes inputs.system
2018-09-11T20:31:10Z I! Loaded aggregators:
2018-09-11T20:31:10Z I! Loaded processors:
2018-09-11T20:31:10Z I! Loaded outputs: prometheus_client
2018-09-11T20:31:10Z I! Tags enabled: host=ip-172-31-17-238
2018-09-11T20:31:10Z I! Agent Config: Interval:10s, Quiet:false, Hostname:"ip-172-31-17-238", Flush Interval:10s
2018-09-11T20:31:30Z D! Output [prometheus_client] buffer fullness: 29 / 10000 metrics.
2018-09-11T20:31:30Z D! Output [prometheus_client] wrote batch of 29 metrics in 763.972µs

Telegraf.conf

# # Read metrics from one or many MongoDB servers:

 [[inputs.mongodb]]
   servers = ["mongodb://telegraf:telegraf@127.0.0.1:27017"]

______________________________________________________

Any possible issues why these counter don't show up at all, i would understand if they displayed 0 but the log as you can see doesn't report much, the user is authenticated as i can retrieve per database counters, but again nothing relating to replica sets.

**Comment by chezneUE** [comment]
Can close this, not a bug with prometheus output, 

However, i do believe this needs to be specified in the documentation. 

Instead use this depending on the number of MongoDB instances you wish to capture metric from.

servers = ["mongodb://telegraf:telegraf@mongodbhere:27017,mongodbhere:27017,mongodbhere:27017/?replicaSet=s0&authSource=admin"]



Maybe should be added to documentation for MongoDB inputs under server configuration

https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mongodb

**Comment by danielnelson** [comment]
@chezneUE Can you summarize what you found the cause to be?

**Comment by chezneUE** [comment]
> @chezneUE Can you summarize what you found the cause to be?

Hello, the issue some is on my part some maybe the documentation, firstly say if you have a primary mongodb and two secondaries.

Create the telegraf users with clustermonitor role, then in the inputs.mongodb of the telegraf configuration add multiple servers such as:


[[inputs.mongodb]]
servers = ["mongodb://telegraf:telegraf@localhost:27017/?replicaSet=s0&authSource=admin"]
                   
[[inputs.mongodb]]
servers = ["mongodb://telegraf:telegraf@10.0.0.1:27017/?replicaSet=s0&authSource=admin"]

[[inputs.mongodb]]
servers = ["mongodb://telegraf:telegraf@10.0.0.2:27017/?replicaSet=s0&authSource=admin"]

You need to specify inputs.mongodb per database in the format above. With source being admin, create users in the admin table and you are good to go.

Aside from that all is working well as it should, however. 

Replication Ops window sec, metric continues to ascend over time. I am not sure what it is but maybe i need to set per database monitoring to true. I will test this tomorrow.

Thanks

**Comment by chezneUE** [comment]
Still not working, i installed telegraf from influxdb, i didn't build the application using Go. 1.7.4 value remains 0.

**Comment by danielnelson** [comment]
Sorry for the long delay, I checked the replication stats today on my replica set and they appear to be working.  Since the `_per_sec` metrics are integers, they must to be greater than 1/sec to register.  However, I recommend ignoring the `_per_sec` metrics and focusing only on the raw counts, it is possible to calculate these rate from the raw counts.

You can also cross reference the reported values using:
```
use admin
db.runCommand({serverStatus:1})
```

Most of the data comes from the metrics, commands section of the response, for example:
```
        "metrics" : {
                "commands" : {
                        "insert" : {
                                "failed" : NumberLong(0),
                                "total" : NumberLong(159)
                        },
```


Many of these values are counters, they don't reset.  You can use the non_negative_derivative function in InfluxDB to display these as rates.

**Comment by edward408-zhang** [comment]
Is there any updates on this issue? how do we set up the config file for replicaset
